### PR TITLE
Use the /menu endpoint when available

### PR DESCRIPTION
**Pull request by ekohl**
Before this patch the pages to check was static. This is problematic because plugins can add pages dynamically. The /menu endpoint returns a JSON representation of the menu available for that user.

This change retrieves the endpoint. When it returns 404, it falls back to the static list. On success the dynamic menu is used.

This depends on https://github.com/theforeman/foreman/pull/6966

The resulting output:
```console
$ pytest --base-url http://localhost:3000 --variables variables.json  --collect-only
================================================== test session starts ===================================================
platform linux -- Python 3.7.5, pytest-5.0.1, py-1.8.0, pluggy-0.12.0
baseurl: http://localhost:3000
sensitiveurl: .* *** WARNING: sensitive url matches http://localhost:3000 ***
rootdir: /home/ekohl/dev/smoker, inifile: pytest.ini
plugins: xdist-1.29.0, variables-1.7.1, base-url-1.4.1, selenium-1.16.0, html-1.21.1, forked-1.0.2, metadata-1.8.0
collected 41 items
<Module tests/test_pages.py>
  <Function test_menu_item[Dashboard]>
  <Function test_menu_item[Facts]>
  <Function test_menu_item[Statistics]>
  <Function test_menu_item[Trends]>
  <Function test_menu_item[Audits]>
  <Function test_menu_item[Config Management]>
  <Function test_menu_item[Report Templates]>
  <Function test_menu_item[All Hosts]>
  <Function test_menu_item[Create Host]>
  <Function test_menu_item[Architectures]>
  <Function test_menu_item[Hardware Models]>
  <Function test_menu_item[Installation Media]>
  <Function test_menu_item[Operating Systems]>
  <Function test_menu_item[Partition Tables]>
  <Function test_menu_item[Provisioning Templates]>
  <Function test_menu_item[Host Groups]>
  <Function test_menu_item[Global Parameters]>
  <Function test_menu_item[Environments]>
  <Function test_menu_item[Classes]>
  <Function test_menu_item[Config Groups]>
  <Function test_menu_item[Smart Variables]>
  <Function test_menu_item[Smart Class Parameters]>
  <Function test_menu_item[Smart Proxies]>
  <Function test_menu_item[Compute Resources]>
  <Function test_menu_item[Compute Profiles]>
  <Function test_menu_item[Subnets]>
  <Function test_menu_item[Domains]>
  <Function test_menu_item[HTTP Proxies]>
  <Function test_menu_item[Realms]>
  <Function test_menu_item[My Account]>
  <Function test_menu_item[Log Out]>
  <Function test_menu_item[Locations]>
  <Function test_menu_item[Organizations]>
  <Function test_menu_item[LDAP Authentication]>
  <Function test_menu_item[Users]>
  <Function test_menu_item[User Groups]>
  <Function test_menu_item[Roles]>
  <Function test_menu_item[Bookmarks]>
  <Function test_menu_item[Settings]>
  <Function test_menu_item[About]>
  <Function test_menu_item[GraphiQL]>

============================================== no tests ran in 0.27 seconds ==============================================
```

### CODE REVIEW

File: tests/test_pages.py [hunk 1]
```diff
@@ -60,22 +61,39 @@ def chrome_options(chrome_options):
...
-    return base_url + request.param
+@pytest.fixture(scope='session')
+def user(variables):
+    return _user(variables)
+
+
+def pytest_generate_tests(metafunc):
+    if 'url' in metafunc.fixturenames:
+        base_url = metafunc.config.option.base_url
+        if not base_url:
```
Comment (evgeni) [inline tests/test_pages.py]: why the `if` instead of just calling `assert` directly?

File: tests/test_pages.py [hunk 2]
```diff
@@ -60,22 +61,39 @@ def chrome_options(chrome_options):
...
+
+        # mimic the variables fixture from pytest-variables
+        user_obj = _user(metafunc.config._variables)  # pylint: disable=protected-access
+
+        response = requests.get(f'{base_url}/menu', auth=(user_obj.username, user_obj.password))
+        if response.status_code == 404:
+            # Menu is only available since Foreman 2.0
+            pages = [base_url + page for page in PAGES]
+        else:
+            assert response
```
Comment (evgeni) [inline tests/test_pages.py]: what is it asserting here?

Comment (ekohl) [reply tests/test_pages.py]: `bool(response)` is equal to `response.ok`
https://github.com/psf/requests/blob/eedd67462819f8dbf8c1c32e77f9070606605231/requests/models.py#L673-L681

File: tests/test_pages.py (hunk 1 above)
Comment (ekohl) [reply tests/test_pages.py]: Good point

File: tests/test_pages.py (hunk 2 above)
Comment (evgeni) [reply tests/test_pages.py]: That's, uh, "interesting". I'm used to raise for status or similar, but yeah, that works too.

**Comment by ekohl** [comment]
Since it's pretty final that the URL is `/menu`, I've marked it as ready for review and will merge it.

**Pull request merged**